src/overlays/actors/ovl_Bg_Relay_Objects/z_bg_relay_objects.c

```diff
@@ -32,23 +43,173 @@ const ActorInit Bg_Relay_Objects_InitVars = {
     (ActorFunc)BgRelayObjects_Update,
     (ActorFunc)BgRelayObjects_Draw,
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Relay_Objects/BgRelayObjects_Init.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Relay_Objects/BgRelayObjects_Destroy.s")
+static InitChainEntry sInitChain[] = {
+    ICHAIN_F32(gravity, 5, ICHAIN_CONTINUE),
+    ICHAIN_VEC3F_DIV1000(scale, 100, ICHAIN_STOP),
+};
+
+void BgRelayObjects_Init(Actor* thisx, GlobalContext* globalCtx) {
+    BgRelayObjects* this = THIS;
+    s32 pad;
+    s32 sp24 = 0;
+    static u32 D_808A9508 = 0;
+
+    Actor_ProcessInitChain(thisx, sInitChain);
+    this->switchFlag = thisx->params & 0x3F;
+    thisx->params = (thisx->params >> 8) & 0xFF;
+    DynaPolyInfo_SetActorMove(&this->dyna, 3);
+    if (thisx->params == WINDMILL_ROTATING_GEAR) {
+        DynaPolyInfo_Alloc(&D_060025FC, &sp24);
+        if (gSaveContext.eventChkInf[6] & 0x20) {
+            thisx->posRot.rot.y = 0x400;
+        } else {
+            thisx->posRot.rot.y = 0x80;
+        }
+        func_800F5718();
+        thisx->room = -1;
+        thisx->flags |= 0x20;
+        if ((D_808A9508 & 2) != 0) {
+            thisx->params = 0xFF;
+            Actor_Kill(thisx);
+        } else {
+            D_808A9508 |= 2;
+            this->actionFunc = func_808A939C;
+        }
+    } else {
+        DynaPolyInfo_Alloc(&D_060003C4, &sp24);
+        if (thisx->room == 0) {
+            this->unk_169 = this->switchFlag - 0x33;
+        } else {
+            this->unk_169 = thisx->room + 1;
+        }
+        thisx->room = -1;
+        this->timer = 1;
+        if (this->unk_169 >= 6) {
+            if (D_808A9508 & 1) {
+                Actor_Kill(thisx);
+            } else {
+                D_808A9508 |= 1;
+                this->actionFunc = BgRelayObjects_Noop;
+            }
+        } else if (this->unk_169 != 5) {
+            Flags_UnsetSwitch(globalCtx, this->switchFlag);
+            if (((1 << this->unk_169) & D_808A9508)) {
```

Comment:
I suggest writing like this if it works
```suggestion
            if (D_808A9508 & (1 << this->unk_169)) {
```